src/test/java/com/google/api/codegen/testdata/java_mock_service_impl_library.baseline

```diff
@@ -297,8 +284,7 @@ public class MockLibraryServiceImpl implements LibraryService  {
   public void updateBookIndex(UpdateBookIndexRequest request,
       StreamObserver<Empty> responseObserver) {
     if (responses != null && !responses.isEmpty()) {
-      Empty response = (Empty) responses.get(0);
-      responses.remove(0);
+      Empty response = (Empty) responses.poll();
```

Comment:
Your logic is correct when the exception is due to an error in the system under test (i.e. the API surface is improperly configured for response testing). If the `responses` queue could be unexpectedly empty due to this kind of user error, then we should be graceful.

OTOH, if the `responses` queue could _only_ be unexpectedly empty due to a bug in the unit test auto-generator (i.e. this `toolkit` code you're working on), then throwing an exception is more appropriate, so that it's obvious that we need to fix the bug rather than an API producer.